xchange-stream-coinbasepro/src/main/java/info/bitrich/xchangestream/coinbasepro/CoinbaseProStreamingExchange.java

```diff
@@ -84,11 +84,17 @@ public Completable disconnect() {
 
   @Override
   public Observable<Throwable> reconnectFailure() {
+    if (streamingService == null)
```

Comment:
Sorry to mess you around, but I've just had another think about this. I think it's perfectly reasonable to expect these observables to fire on initial connection, not just on automated reconnections.  It seems wrong that we can't register to hear about connections/reconnections until _after_ we've connected. Could we just return publishers here and hook them up to the `StreamingService` before connecting?